hera_qm/firstcal_metrics.py

```diff
@@ -0,0 +1,544 @@
+            if plot_type == 'both':
+                axes = ax
+                ax = axes[1]
+            plabel = []
+            ax.grid(True)
+            for i, index in enumerate(plot_ants):
+                p, = ax.plot(self.frac_JD, self.delay_offsets[index],
+                             marker='.', c=cmap[i], **plt_kwargs)
+                plabel.append(p)
+            ax.set_xlabel('fraction of JD %d' % self.start_JD, fontsize=14)
+            ax.set_ylabel('delay offset [ns]', fontsize=14)
+            if plot_type == 'both':
+                ax = axes
+
+        # add legend
+        if custom_ax is False:
+            ax = fig.add_axes([1.0, 0.1, 0.05, 0.8])
+            ax.axis('off')
+            ax.legend(plabel, [self.ants[i] for i in plot_ants])
+        else:
+            ax[-1].legend(plabel, [self.ants[i] for i in plot_ants])
+
+        if save is True and custom_ax is False:
+            if fname is None:
+                fname = self.file_stem + '.dlys.png'
+            fig.savefig(fname, bbox_inches='tight')
+
+    def plot_zscores(self, fname=None, plot_type='full', cm='viridis_r', ax=None, figsize=(10, 6),
+                     save=False, kwargs={'cmap': 'viridis_r', 'vmin': 0, 'vmax': 5}):
+        """
+        Plot antenna delay solution z_scores
+
+        Input:
+        ------
+
+        fname : str, default=None
+            filename
+
+        plot_type : str, default='full'
+            Type of plot to make
+            'full' : plot zscore for each (N_ant, N_times)
+            'time_avg' : plot zscore for each (N_ant,) avg over time
+
+        cm : str, default='viridis_r'
+            colormap
+
+        ax : axis object, default=None
+            matplotlib axis object
+
+        figsize : tuple, default=(10,6)
+            figsize if creating figure
+
+        save : bool, default=False
+            save figure to file
+
+        kwargs : dict
+            plotting kwargs
+        """
+
+        custom_ax = True
+        if ax is None:
+            custom_ax = False
+            fig = plt.figure(figsize=figsize)
+            ax = fig.add_subplot(111)
+
+        # Get zscores
+        ant_avg, ant_std, time_std, agg_std, z_scores = self.delay_std()
+
+        # Plot zscores
+        if plot_type == 'full':
+            # plot
+            xlen = [self.frac_JD.min(), self.frac_JD.max()]
+            xlen_round = [np.ceil(self.frac_JD.min() * 1000) / 1000,
+                          np.floor(self.frac_JD.max() * 1000) / 1000]
+            ylen = [0, len(self.ants)]
+            cax = ax.matshow(z_scores, origin='lower', aspect='auto',
+                             extent=[xlen[0], xlen[1], ylen[0], ylen[1]], **kwargs)
+
+            # define ticks
+            ax.xaxis.set_ticks_position('bottom')
+            xticks = np.arange(xlen_round[0], xlen_round[1] + 1e-5, 0.001)
+            ax.set_xticks(xticks)
+            # [t.set_rotation(20) for t in ax.get_xticklabels()]
```

Comment:
If this line is not needed, it should be removed instead of being commented out.